pkg/scyllaclient/client.go

```diff
@@ -185,6 +185,75 @@ func (c *Client) GetLocalHostId(ctx context.Context, host string, retry bool) (s
 	return resp.GetPayload(), nil
 }
 
+func (c *Client) GetIPToHostIDMap(ctx context.Context, host string) (map[string]string, error) {
+	if len(host) > 0 {
+		ctx = forceHost(ctx, host)
+	}
+
+	resp, err := c.scyllaOps.StorageServiceHostIDGet(&scyllaOperations.StorageServiceHostIDGetParams{
+		Context: ctx,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	mapping := make(map[string]string, len(resp.Payload))
+
+	for _, e := range resp.Payload {
+		mapping[e.Key] = e.Value
+	}
+
+	return mapping, nil
+}
+
+func (c *Client) GetTokenRing(ctx context.Context, host string) ([]string, error) {
+	if len(host) > 0 {
+		ctx = forceHost(ctx, host)
+	}
+
+	resp, err := c.scyllaOps.StorageServiceTokensGet(&scyllaOperations.StorageServiceTokensGetParams{Context: ctx})
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.GetPayload(), nil
+}
+
+func (c *Client) GetNodeTokens(ctx context.Context, host, endpoint string) ([]string, error) {
+	ctx = forceHost(ctx, host)
+
+	resp, err := c.scyllaOps.StorageServiceTokensByEndpointGet(&scyllaOperations.StorageServiceTokensByEndpointGetParams{
+		Endpoint: endpoint,
+		Context:  ctx,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.GetPayload(), nil
+}
+
+func (c *Client) Cleanup(ctx context.Context, host string, keyspace string) error {
+	const (
+		// Cleanup is synchronous call and may take a long time to finish.
+		// Default request timeout in client is not big enough to clean huge keyspace.
+		cleanupTimeout = 24 * time.Hour
+	)
+
+	ctx = forceHost(ctx, host)
+	ctx = customTimeout(ctx, cleanupTimeout)
```

Comment:
to me this seems like it should be set by the caller and the reason context is being passed through. I don't see a wy for the user to raise it to 48h this way

Reply:
it follows how scyllaclient is currently designed. Changing it is out of scope of this PR.